ecommerce/frontend/src/components/SignUpForm.js

```diff
@@ -0,0 +1,289 @@
+            <Form.Item
+                name="password"
+                label="Password"
+                rules={[
+                    {
+                        required: true,
+                        message: 'Please input your password!',
+                    },
+                ]}
+                hasFeedback
+            >
+                <Input.Password />
+            </Form.Item>
+            
+            {/* Password confirmation input */}
+            <Form.Item
+                name="confirm"
+                label="Confirm Password"
+                dependencies={['password']}
+                hasFeedback
+                rules={[
+                    {
+                        required: true,
+                        message: 'Please confirm your password!',
+                    },
+                    ({ getFieldValue }) => ({
+                        validator(rule, value) {
+                            if (!value || getFieldValue('password') === value) {
+                                return Promise.resolve();
+                            }
+
+                            return Promise.reject('The two passwords that you entered do not match!');
+                        },
+                    }),
+                ]}
+            >
+                <Input.Password />
+            </Form.Item>
+
+            {/* City input */}
+            <Form.Item
+                name={['address', 'city']}
+                label="City"
+                rules={[
+                    {
+                        required: true,
+                        message: 'Please select your city!',
+                    },
+                ]}
+            >
+                <Input />
+            </Form.Item>
+            
+            {/* State input */}
+            <Form.Item
+                name={['address', 'state']}
+                label="State"
+                rules={[
+                    {
+                        required: true,
+                        message: 'Please select your state!',
+                    },
+                ]}
+            >
+                <Input />
+            </Form.Item>
+
+            {/* Zip code input */}
+            <Form.Item
+                name={['address', 'zipcode']}
+                label="Zip Code"
+                rules={[
+                    {
+                        required: true,
+                        message: 'Please enter your zip code!',
+                    },
+                ]}
+            >
+                <Input />
+            </Form.Item>
+            
+            {/* Main address input */}
+            <Form.Item
+                name={['address', 'mainaddress']}
+                label="Main Address"
+                rules={[
+                    {
+                        required: true,
+                        message: 'Please enter your address!',
+                    },
+                ]}
+            >
+                <Input />
+            </Form.Item>
```

Comment:
1. for keys please use camelCase (example mainaddress -> mainAddress, zipcode -> zipCode). It is what we use in JS.
2.

**1.1.1.2.6** Location (not mandatory for customer and admin users)

The location information is not mandatory for customer users. Therefore the `required: true` needs to depend on whether the Guest user is registering as a Customer or Vendor. This might difficult since required needs to react to value change of a form item so I can help you with it when you are available.

Reply:
1. I generally automatically use camelCase but I missed those, ill correct it. 
2. Maybe we can set a `onChange` callback on the radio buttons and define a state like `locationEnabled` and make it both disable the input boxes and set `required: locationEnabled`. Will that work?

Reply:
corrected the non camelCase stuff on 4th commit

Reply:
for 2:

That sounds like it could work actually. Can you try it? If it still doesn't work then let me know.